app/controllers/piecharts_controller.rb

```
@@ -0,0 +1,26 @@
+import Reports;
+
+class PiechartsController < ApplicationController
+  def index
+    begin   
+      filter ||= {'group_by' => 'test.name'}
+      filter['until'] = Date.parse(params['bar_date']).iso8601
+      filter['since'] = (Date.parse(params['bar_date'])-1.month).iso8601
+       
+     #TODO      options= {"date_range"=>{"start_time"=>{"gte"=>Date.parse((Date.parse(params['bar_date'])-1.month).iso8601, "lte"=>:Date.parse(params['bar_date']).iso8601}}} 
+     #TODO       some_value = Reports::Grouped.by_name(current_user, nil, options)
+    
+      results = TestResult.query(filter, current_user).execute
+      sorted=results['tests'].map do |test|
+      {
+        label: test['test.name'],
+        value: test['count']
+      }
+     end 
+     render json: sorted 
+   rescue Exception => e 
```

Shouldn't really catch all exceptions unless you re-raise them after doing your logging or whatever